pbs.py

```diff
@@ -13,57 +14,86 @@
-            os.chdir(HOMEPATH + 'qt')
-        finally:
-            sys.exit(0)
+        sys.exit(0)
 
 
-def make_pbsnodes_yaml(orig_file, yaml_file):
+def make_pbsnodes(orig_file, out_file, write_method):
     """
-    reads PBSNODES_ORIG_FILE sequentially and puts its information into a new yaml file
+    reads PBSNODES_ORIG_FN sequentially and puts its information into a new yaml file
     """
     check_empty_file(orig_file)
-    blocks = read_all_blocks(orig_file)
-    with open(yaml_file, 'w') as fout:
-        # lastcore = ''
-        for block in blocks:
-            fout.write('domainname: ' + block['Domain name'] + '\n')
-
-            nextchar = block['state'][0]
-            state = "'-'" if nextchar == 'f' else nextchar
-            fout.write('state: ' + state + '\n')
-
-            fout.write('np: ' + block['np'] + '\n')
-            if block.get('gpus') > 0:  # this should be rare.
-                fout.write('gpus: ' + block['gpus'] + '\n')
-            try:  # this should turn up more often, hence the try/except.
-                _ = block['jobs']
-            except KeyError:
-                pass
-            else:
-                write_jobs_cores(block['jobs'], fout)
-            fout.write('---\n')
+    raw_blocks = _read_all_blocks(orig_file)
+    all_pbs_values = []
+    for block in raw_blocks:
+        pbs_values = dict()
+        pbs_values['domainname'] = block['domainname']
+
+        nextchar = block['state'][0]
+        state = (nextchar == 'f') and "-" or nextchar
+
+        pbs_values['state'] = state
+        pbs_values['np'] = block['np']
+
+        if block.get('gpus') > 0:  # this should be rare.
+            pbs_values['gpus'] = block['gpus']
+        try:  # this should turn up more often, hence the try/except.
+            _ = block['jobs']
+        except KeyError:
+            pass
+        else:
+            pbs_values['core_job_map'] = []
+            jobs = block['jobs'].split(',')
+            for job, core in get_jobs_cores(jobs):
+                _d = {}
+                _d['job'] = job
+                _d['core'] = core
+                pbs_values['core_job_map'].append(_d)
+            all_pbs_values.append(pbs_values)
+    pbs_dump_all(all_pbs_values, out_file, pbsnodes_mapping[write_method])
+
+
+def pbsnodes_write_lines(l, fout):
+    for _block in l:
+        fout.write('---\n')
+        fout.write('domainname: ' + _block['domainname'] + '\n')
+        fout.write('state: ' + "'" + _block['state'] + "'" + '\n')
+        fout.write('np: ' + _block['np'] + '\n')
+        if _block.get('gpus') > 0:
+            fout.write('gpus: ' + _block['gpus'] + '\n')
+        try:  # this should turn up more often, hence the try/except.
+            core_job_map = _block['core_job_map']
+        except KeyError:
+            pass
+        else:
+            _write_jobs_cores(core_job_map, fout)
+        fout.write('...\n')
 
 
-def write_jobs_cores(jobs, fout):
+def _write_jobs_cores(job_cores, fout):

```

Comment:
i have prepended an underscore in those functions which are supposed to be used only within the module